### ISSUE TITLE
Streaming API — incremental fit/transform for datasets exceeding RAM

### ISSUE THREAD

**Issue by DeathSurfing** [featrs/featrs#80]
## Description

Add a **streaming API** that supports incremental (batched) `fit` and `transform` for datasets that exceed available memory. Instead of requiring the entire input DataFrame in RAM, the API accepts an iterator/reader of DataFrames (batches) and processes them one batch at a time, accumulating statistics without holding all data at once.

This complements LazyFrame Support (#33) — Lazy enables query optimization and partial materialization, but `collect()` may still materialize intermediate results. Streaming goes further by guaranteeing batch-by-batch execution.

This is **High Priority** infrastructure — depends on LazyFrame Support (#33). Useful for:
- Log processing with terabytes of data
- Streaming ML (continuous training updates)
- The "Online Statistics" feature (#35)

## Proposed API

```rust
pub trait FitStreaming {
    type Output;
    
    /// Begin streaming fit. Returns a builder that can accept batches.
    fn begin_fit(&mut self) -> Result<StreamingFitter<'_, Self>>;
}

pub struct StreamingFitter<'a, T> {
    transformer: &'a mut T,
    state: StreamingState,
    n_samples_seen: u64,
    n_batches: u64,
}

impl<'a, T> StreamingFitter<'a, T> {
    /// Ingest a batch. Updates running statistics without holding prior batches.
    pub fn partial_fit(&mut self, batch: DataFrame) -> Result<()>;
    
    /// Finalize the fit (may compute derived parameters from running statistics).
    pub fn finalize(self) -> Result<()>;
}

// Or simpler: extend existing Fit
pub trait FitPartial {
    fn partial_fit(&mut self, x: DataFrame) -> Result<()>;
    fn finalize_fit(&mut self) -> Result<()>;
}

// Transform on stream
pub trait TransformStreaming {
    fn transform_streaming(&self, batches: impl Iterator<Item = DataFrame>) -> Result<Vec<DataFrame>>;
    // or -> impl Iterator<Item = Result<DataFrame>>
}
```

### Approach

1. Create `src/traits/streaming.rs` with `FitPartial` (and maybe `TransformStreaming`)
2. For each transformer that can support streaming fit, implement `FitPartial`:
   - **StandardScaler, MinMaxScaler, MaxAbsScaler, VarianceThreshold**: maintain running mean/std/min/max/count using Welford's online algorithm (mean + M2 sum, supports incremental update)
   - **SimpleImputer**: accumulating counts and sums for mean/median; for median, must hold all values or use an approximate streaming quantile algorithm (T-Digest, Greenwald-Khanna)
   - **RobustScaler**: needs quantiles → same streaming quantile challenge as Imputer
   - **QuantileTransformer, KBinsDiscretizer (quantile strategy), Winsorizer, OutlierClipper**: same — streaming quantiles required
   - **PowerTransformer**: MLE over lambdas can use gradient descent incrementally over batches
   - **TargetEncoder, LeaveOneOutEncoder**: accumulating per-category target sum and count — trivially streaming
3. For `TransformStreaming`:
   - Most transforms are stateless folds over rows — trivially streamable row-by-row (or batch-by-batch)
4. Pipeline integration:
   - `Pipeline::partial_fit` iterates batches through each step, accumulating each step's partial-fit state
   - `Pipeline::transform_streaming` applies each step's transform per batch, streaming through
5. Streaming Quantile Algorithm: T-Digest (Crate: `tdigest`) is a good standard. Otherwise Greenwald-Khanna. Adds a `tdigest` dependency (small, well-maintained).

## Edge Cases

- **Empty batches**: skip without error
- **Single batch**: equivalent to non-streaming fit
- **Batch order**: don't assume order-independence for skewed distributions; document that batch size can affect tie-breaking / quantile estimates slightly (inherent in streaming estimation — give epsilon guarantees)
- **Resume after finalize**: should reject `partial_fit` after `finalize_fit` unless the algorithm supports warmwake (re-fit)
- **Partial fit before previous finalize**: error or auto-finalize; recommend explicit sequence
- **Different dtypes across batches**: error on schema mismatch (link to SchemaValidator #38)
- **Numerical stability**: Welford's algorithm is standard; document that the running mean + M2 are numerically stable for arbitrary order
- **Column count consistency**: errors if a batch has different columns than expected

## Testing

- Feed 100 batches through `partial_fit`; verify final statistics match fitting on a single concatenated DataFrame (with appropriate epsilon for floating-point)
- Compare streaming StdScaler result to eager StdScaler on identical data: proptest (#22)
- T-Digest median: approximate, with relative error within `1e-3` for terabyte test data (skip terabyte tests — use small batches with re-fitted closure)
- Empty batch handling
- Single batch equivalence to non-streaming
- Transform streaming: batches come out with each transform correctly applied; combined output identical to non-streaming equivalent on the same data

## Dependencies

- `tdigest` (or similar streaming-quantile crate) — new dependency, required for streaming RobustScaler / QuantileTransformer / median imputation
- Welford's algorithm: no dep — inline implementation

## Related Issues

- #33 — LazyFrame Support (prerequisite — streaming transformers should emit lazy exprs when possible)
- #35 — Online Statistics (uses streaming APIs — actually this issue and #35 heavily overlap, see #35 for the running-stats object abstraction)
- #21 — Criterion benchmarks (compare streaming vs eager for memory and throughput)
- #24 — Polars-native kernels (Polars already has streaming support; Lean on `LazyFrame::collect Streaming` when possible)
- #38 — SchemaValidator (validate each batch's schema)
- HIGH PRIORITY (depends on #33)
